2.7/Dockerfile

```diff
@@ -46,7 +46,12 @@ RUN set -ex \
             && wget -O /tmp/get-pip.py 'https://bootstrap.pypa.io/get-pip.py' \
             && python2 /tmp/get-pip.py "pip==$PYTHON_PIP_VERSION" \
             && rm /tmp/get-pip.py \
-        && pip install --no-cache-dir --upgrade pip==$PYTHON_PIP_VERSION \
+# we use "--force-reinstall" for the case where the version of pip we're trying to install is the same as the version bundled with Python
+# ("Requirement already up-to-date: pip==8.1.2 in /usr/local/lib/python3.6/site-packages")
+# https://github.com/docker-library/python/pull/143#issuecomment-241032683
+        && pip install --no-cache-dir --upgrade --force-reinstall "pip==$PYTHON_PIP_VERSION" \
+# then we use "pip list" to ensure we don't have more than one pip version installed
+# https://github.com/docker-library/python/pull/100
         && [ "$(pip list |tac|tac| awk -F '[ ()]+' '$1 == "pip" { print $2; exit }')" = "$PYTHON_PIP_VERSION" ] \
```

Comment:
It seems they are...
Last I checked the Dockerfile parser was very simplistic (although it seems to have got slightly fancier in 1.12). It basically just checks for a `#` at the start of a line (the only kind of supported comment) and ignores that line.